internal/executor/metrics/metrics.go

```diff
@@ -56,6 +75,11 @@ func Run(ctx context.Context) (chan *api.ResourceUtilization, chan error) {
 				return
 			}
 
+			if logger != nil {
+				logger.Infof("CPUs used: %.2f, CPU usage: %.2f%%, memory used: %s", numCpusUsed, numCpusUsed*100.0,
```

Comment:
Maybe use debug level?

Reply:
> Quick question. Have you looked at `gopsutil`'s `docker.CgroupCPUDocker`. Seems if you pass `self` as `containerId` it might work similar... `cgroup2` might not be supported though.

* it doesn't provide any information in regard to memory used by a given control group
* it doesn't compute an absolute value of how many CPUs were used by a given control group
* as you've mentioned, no cgroup version 2 support, [which is slowly gaining traction](https://medium.com/nttlabs/cgroup-v2-596d035be4d7)
* it only supports Docker-style cgroup names (https://github.com/shirou/gopsutil/blob/5268453c1e25853698065e9e5cb14afc175610c3/v3/docker/docker_linux.go#L279-L292) are supported, which wouldn't work on GKE, for example
* it has a very naive subsystem discovery implementation (it assumes that the all `cgroup` subsystems are mounted to `/sys/fs/cgroup`, which may not be the case)

Reply:
I don't think this matters much since the `logger` is only set in `cmd/metrics/metrics.go`, which is a special command used to test this.